metricflow/naming/linkable_spec_name.py

```diff
@@ -37,8 +37,13 @@ def from_name(qualified_name: str) -> StructuredLinkableSpecName:
         if len(name_parts) == 1:
             return StructuredLinkableSpecName(entity_link_names=(), element_name=name_parts[0])
 
+        for date_part in DatePart:
+            if name_parts[-1] == StructuredLinkableSpecName.date_part_suffix(date_part=date_part):
+                raise ValueError(
+                    "Dunder syntax not supported for querying date_part. Use `group_by` object syntax instead."
+                )
+
```

Comment:
Nice! Now if we mess up we'll get exceptions and we can restructure the interfaces at that time.